src/ai/backend/manager/models/endpoint.py

```diff
@@ -494,6 +583,95 @@ async def get(
         return row
 
 
+class EndpointAutoScalingRuleRow(Base):
+    __tablename__ = "endpoint_auto_scaling_rules"
+
+    id = IDColumn()
+    metric_source = sa.Column("metric_source", StrEnumType(AutoScalingMetricSource), nullable=False)
+    metric_name = sa.Column("metric_name", sa.Text(), nullable=False)
+    threshold = sa.Column("threshold", DecimalType(), nullable=False)
+    comparator = sa.Column("comparator", StrEnumType(AutoScalingMetricComparator), nullable=False)
+    step_size = sa.Column("step_size", sa.Integer(), nullable=False)
+    cooldown_seconds = sa.Column("cooldown_seconds", sa.Integer(), nullable=False, default=300)
+
+    min_replicas = sa.Column("min_replicas", sa.Integer(), nullable=True)
+    max_replicas = sa.Column("min_replicas", sa.Integer(), nullable=True)
+
+    created_at = sa.Column(
+        "created_at",
+        sa.DateTime(timezone=True),
+        server_default=sa.text("now()"),
+        nullable=True,
+    )
+    last_triggered_at = sa.Column(
+        "last_triggered_at",
+        sa.DateTime(timezone=True),
+        nullable=True,
+    )
+
+    endpoint = sa.Column(
+        "endpoint",
+        GUID,
+        sa.ForeignKey("endpoints.id", ondelete="CASCADE"),
+        nullable=False,
+    )
+
+    endpoint_row = relationship("EndpointRow", back_populates="endpoint_auto_scaling_rules")
+
+    @classmethod
+    async def list(
+        cls, session: AsyncSession, load_endpoint=False
+    ) -> list["EndpointAutoScalingRuleRow"]:
+        query = sa.select(EndpointAutoScalingRuleRow)
+        if load_endpoint:
+            query = query.options(selectinload(EndpointAutoScalingRuleRow.endpoint_row))
+        result = await session.execute(query)
+        return result.scalars().all()
+
+    @classmethod
+    async def get(
+        cls, session: AsyncSession, id: uuid.UUID, load_endpoint=False
+    ) -> "EndpointAutoScalingRuleRow":
+        query = sa.select(EndpointAutoScalingRuleRow).filter(EndpointAutoScalingRuleRow.id == id)
+        if load_endpoint:
+            query = query.options(selectinload(EndpointAutoScalingRuleRow.endpoint_row))
+        result = await session.execute(query)
+        row = result.scalar()
+        if not row:
+            raise ObjectNotFound("endpoint_auto_scaling_rule")
+        return row
+
+    def __init__(
```

Comment:
Is this constructor strictly required as it's already an ORM class?

Reply:
The purpose of this constructor is to avoid manually assigning individual properties after creating the object.

Reply:
I think it should just work even without this method.

To avoid boilerplate constructor methods for SQLAlchemy models, we could consider using [SQLModel](https://sqlmodel.tiangolo.com/) and similar libraries. I'm going to leave this as a future exploration.